tests/mock_dcgm_agent.py

```diff
@@ -0,0 +1,208 @@
+# Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from unittest.mock import MagicMock
+from collections import defaultdict
+
+import model_analyzer.monitor.dcgm.dcgm_structs as structs
+import model_analyzer.monitor.dcgm.dcgm_fields as dcgm_fields
+from model_analyzer.model_analyzer_exceptions import TritonModelAnalyzerException
+
+
+class MockDCGMAgent:
+
+    device_groups = {}
+    field_groups = {}
+    devices = [{
+        'pci_bus_id': b'00000000:65:00.0',
```

Comment:
They are dummy values but they need to follow a certain format.

Reply:
Instead of having magic numbers, can they be defined somewhere?
```
TEST_PCI_BUS_ID = b'00000000:65:00.0'
TEST_UUID = b'd02ae7ed-4471-7046-4215-fad351e5fc10'

...

devices = [{ 'pci_bus_id' : TEST_PCI_BUS_ID, 'uuid' : TEST_UUID }]
```

Reply:
In general, you want to avoid magic numbers, and lean towards named constants. It encourages reuse, provides a single place to change the values in the future, and improves the readability and purpose of the number.